### PR TITLE
Events work blog

### PR DESCRIPTION
**Pull request by scpaye506**

### CODE REVIEW

File: components/Hamburger/index.js [hunk 1]
```diff
@@ -6,16 +6,13 @@ import {nameToHex} from 'nameToColor';
 class Hamburger extends Component {
   render(): ?ReactElement {
     var className = 'Hamburger';
-    var {target, onClick, color} = this.props;
-    var {topOfViewport, navVisible} = this.context;
+    var {target, onClick, color, overlayVisible} = this.props;
```
Comment (iamdustan) [inline components/Hamburger/index.js]: update PropTypes for this?

File: components/Hamburger/index.js [hunk 2]
```diff
@@ -6,16 +6,13 @@ import {nameToHex} from 'nameToColor';
...
     var hamburgerClass = this.props.color === 'yellow' ? ' is-black' : '';
-    var backgroundColor = nameToHex(this.props.color);
 
-    if (navVisible) className = className + ' is-x';
-    if (!topOfViewport) className = className + ' is-floating';
+    if (overlayVisible) className = className + ' is-x';
 
     return (
-      <a style={{backgroundColor: !topOfViewport && backgroundColor, ...this.props.style}} href={target} className={className} onClick={this.context.toggleNav}>
+      <a href={target} className={className} onClick={this.props.onClick}>
```
Comment (iamdustan) [inline components/Hamburger/index.js]: could you use the locally scoped `onClick` already?

File: components/Navigation/index.js [hunk 3]
```diff
@@ -22,40 +22,93 @@ var SECONDARY = [
...
+  componentDidMount(): void {
+    if (typeof window === 'undefined') return;
+    if (window.scrollY) {
+      this.setState({atTop: false});
+    }
+    window.addEventListener('scroll', _ => {
+      var showNav = true;
+      var atTop = false;
+
+      if (this.previousScrollPos - window.scrollY < 0) {
```
Comment (iamdustan) [inline components/Navigation/index.js]: ugh. Are we doing the disappearing/reappearing nav thing with this?

File: components/Navigation/index.js [hunk 4]
```diff
@@ -22,40 +22,93 @@ var SECONDARY = [
...
+  }
+
+  componentWillUnmount(): void {
+    if (typeof window === 'undefined') return;
+    window.removeEventListener('scroll');
+  }
+
+  toggleOverlay(): void {
+    if (!this.state.overlayVisible) {
+      document.getElementsByTagName('body')[0].className = 'noscroll';
```
Comment (iamdustan) [inline components/Navigation/index.js]: this can just be `document.body`

File: components/Navigation/index.js [hunk 5]
```diff
@@ -22,40 +22,93 @@ var SECONDARY = [
...
+    this.setState({overlayVisible: !this.state.overlayVisible});
+  }
+
   render(): ?ReactElement {
-    var {visible} = this.props;
-    var className = `Navigation-mobile ${visible ? 'is-visible' : 'is-not-visible'}`;
+    var mobileClassName = `Navigation-mobile ${this.state.overlayVisible ? 'is-visible' : 'is-not-visible'}`;
+    var mainClass = `Navigation-main ${this.state.showNav ? 'is-visible' : 'is-not-visible'}`;
+    mainClass += this.state.atTop ? ' is-top' : '';
+    var hamburgerClass = `Navigation-hamburger ${this.state.showNav ? 'is-visible' : 'is-not-visible'}`;
```
Comment (iamdustan) [inline components/Navigation/index.js]: you could probably do the `var somethingSomething = this.state.showNav ? 'is-visible' : 'is-not-visible'` and not do so many if else’s

File: handlers/BlogArticle/index.js [hunk 6]
```diff
@@ -48,9 +48,10 @@ class BlogArticle extends React.Component {
     return (
       <article className="BlogArticle">
         <Hero
-          title={title}
+          title="Blog"
```
Comment (iamdustan) [inline handlers/BlogArticle/index.js]: this is going to make for really wonky header hierarchies

File: components/Navigation/index.js (hunk 3 above)
Comment (scpaye506) [inline components/Navigation/index.js]: for now! I'm going to look into some better things at  least for mobile :)

File: components/Hamburger/index.js (hunk 2 above)
Comment (scpaye506) [inline components/Hamburger/index.js]: the Navigation component is passing it the onClick event now and Navigation is the only component using hamburger now.

**Pull request merged**